### PR TITLE
Implemented train mode

### LINKED ISSUE #2: Implement 'train' mode
Implement "train" functionality. It should be able to take hyperparameters and dataset as an input, decide which algorithm implementation to use, and run a training loop.

Blocked by #4, #5

### PR DESCRIPTION
**Pull request by dijana-z**
Closes #2

### CODE REVIEW

**Review by stefanpantic — APPROVED** [review]
Amazing!
